rfcs/0007_reorganize_repositories.md

```diff
@@ -0,0 +1,74 @@
+- Circle-CI does not include style or unit-test checks on new code. 
+- When merging a discontinued project, PR review does not need to be as thorough as for code pushed to the main `tbp.monty` repository.
+- Not intended as a code-base that community members can contribute to. As such, issues like bugs are less likely to be noticed and fixed. 
+
+We could then have two additional types of repositories, with no predefined number of these. These repository *types* would be:
+
+`tbp_package.name_of_package`
+- E.g. `tbp_package.floppy`
+- A package that is a collection of modules that are intended to be re-used in other projects. This does not imply that it is a package that will be maintained long-term for a broader community, but is rather something that we see ourselves re-using at the TBP.
+- Should be well-documented, and have some unit-test coverage and style checks. PR reviews should be of a similar standard to `tbp.monty`.
+- Open to contributions from the community.
+
+`tbp_paper.name_of_paper`
+- E.g. `tbp_paper.dmc_paper`
+- A repository that contains the code required to replicate results and figures from a paper.
+- A given paper can be broken up into multiple repositories if it makes use of highly distinct frameworks for different parts of the paper, such as Pytorch vs. Monty.
+- PR reviews should be of a similar standard to `tbp.monty`.
+
+More concretely, the structure that would be created given our current codebase and work on the DMC paper is as follows:
+
+```
+tbp.monty/
+monty_lab/
+tbp_package.floppy/
+tbp_paper.dmc_monty/  # Most of the code for the DMC paper, including for generating the figures
+tbp_paper.dmc_pytorch/  # Code to generate the results for the Pytorch models used in the paper
+```
+
+### Other Guidance
+- When opening a PR, if there is any doubt about the best destination for the code, it is best to discuss this with the team.
+- We should have a relatively low threshold for creating new repositories where appropriate.
+
+# Motivation
+
+As the Thousand Brains Project grows, it will be important to have a better way of organizing the codebase. This RFC proposes we establish a sustainable structure that ensures code remains high quality and accessible, while minimizing overhead for contributors.
+
+# Open Questions
+
+- If a given paper is separated into multiple repositories, will this create unecessary challenges for someone who wants to replicate the results (e.g. generating a figure)? We can think about how to structure results folders to make this as easy as possible.
+- Should we be open to merging code into `monty_lab`'s `main` as a form of staging before it is eventually moved to a final repository? This would be akin to what has organically happened with the DMC work.
+    - An advantage of this would be to make it easier for maintainers to review provisional code that is not yet ready for a final repository.
```

Comment:
I would tend not to do that. There is no problem with starting an empty repository and adding slowly to it over time. If it is a separate repo we can add style and unit test check from the start to make sure quality standards are adhered to.

Reply:
Thanks yeah good point, I'll leave this open for now during the discussion.